src/services/rollouts.js

```diff
@@ -114,12 +114,117 @@ module.exports = class RolloutsHelper {
 
 			return responses.successResponse({
 				statusCode: httpStatusCode.ok,
-				message: 'ROLLOUT_CREATED_SUCCESSFULLY',
+				message: 'ROLLOUT_SAVED_SUCCESSFULLY',
 				result: { id: rolloutCreate.id },
 			})
 		} catch (error) {
 			await transaction.rollback() // Rollback transaction on any error
 			throw error
 		}
 	}
+
+	/**
+	 * Rollout update
+	 * @method
+	 * @name update
+	 * @param {Integer} rolloutId - Rollout Id.
+	 * @param {Object} bodyData - request data.
+	 * @param {String} loggedInUserId - userId
+	 * @param {String} orgId - organization id
+	 * @returns {JSON} - rollout update response.
+	 */
+
+	static async update(rolloutId, bodyData, loggedInUserId, orgId) {
+		try {
+			let rollout = await rolloutQueries.findOne({
+				id: rolloutId,
+				organization_id: orgId,
+			})
+
+			if (!rollout?.id) {
+				return responses.failureResponse({
+					message: 'ROLLOUT_NOT_FOUND',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			// Prevent changes to the resource id if a rollout is published
+			if (bodyData.resource_id && bodyData.resource_id != rollout.resource_id) {
+				if (rollout.rollout_date || bodyData.published_id) {
+					return responses.failureResponse({
+						message: 'CANT_CHANGE_RESOURCE',
+						statusCode: httpStatusCode.bad_request,
+						responseCode: 'CLIENT_ERROR',
+					})
+				}
+
+				let resource = await resourceQueries.findOne({
+					id: bodyData.resource_id,
+					organization_id: orgId,
+					stage: common.RESOURCE_STAGE_COMPLETION,
+				})
+
+				if (!resource?.id) {
+					return responses.failureResponse({
+						message: 'RESOURCE_NOT_FOUND',
+						statusCode: httpStatusCode.bad_request,
+						responseCode: 'CLIENT_ERROR',
+					})
+				}
+			}
+
+			bodyData = _.omit(bodyData, ['id', 'resource_type', 'type', 'organization_id', 'user_id'])
+
+			if (bodyData.targeting_criteria) {
+				const fileName = `${loggedInUserId}${rolloutId}rollout.json`
+				const rolloutUploadStatus = await resourceService.uploadToCloud(
```

Comment:
@priyanka-TL can you share sample complete URL in cloud for rollout file and also the resource cloud file that will be stored?

Reply:
rollout/314/28/2b8af968-18a0-4110-a6a1-19b58178bc34/31428rollout.json
this we are storing in db 
complete url : https://mentoring-prod-storage.s3.ap-south-1.amazonaws.com/rollout/314/28/2b8af968-18a0-4110-a6a1-19b58178bc34/31428rollout.json